daemon/loop.go

```diff
@@ -22,7 +23,8 @@ const (
 // Loop for potentially long-running stuff. This includes running
 // jobs, and looking for new commits.
 
-func (d *Daemon) Loop(stop chan struct{}, logger log.Logger) {
+func (d *Daemon) Loop(stop chan struct{}, wg *sync2.WaitGroup, logger log.Logger) {
+	defer wg.Done()
```

Comment:
Can't do this, this would case a race. I.e. no guarantee that `wg.Add()` inside the goroutine would run be for the outer `wg.wait()`.

(I only realised this by trying it out! :-) )